ivy/functional/frontends/numpy/statistics/correlating.py

```diff
@@ -0,0 +1,27 @@
+import ivy
+import numpy as np
+
+
+def sum(
+    x,
+    /,
+    *,
+    axis=None,
+    dtype=None,
+    keepdims=False,
+    out=None,
+    initial=np._NoValue,
```

Comment:
You should use None here instead of np._NoValue as these frontend functions should not have dependencies to those other frameworks.

Reply:
Hi @Daniel4078 , **None** is not same as **np._NoValue**, I have tested in my jupyter notebook

```
import numpy as np


a = np.arange(8).reshape((2, 2, 2))
print(a.shape)
print("a:\n", a)

b = np.sum(a, keepdims=False, axis=0)
print("b:\n", b)

b = np.sum(a, keepdims=False, axis=0, initial=np._NoValue, where=np._NoValue)
print("b:\n", b)
```

- bad case
```
import numpy as np


a = np.arange(8).reshape((2, 2, 2))
print(a.shape)
print("a:\n", a)

b = np.sum(a, keepdims=False, axis=0)
print("b:\n", b)

b = np.sum(a, keepdims=False, axis=0, initial=None, where=None)
print("b:\n", b)

```
with errors
```
ValueError                                Traceback (most recent call last)
Input In [3], in <cell line: 11>()
      8 b = np.sum(a, keepdims=False, axis=0)
      9 print("b:\n", b)
---> 11 b = np.sum(a, keepdims=False, axis=0, initial=None, where=None)
     12 print("b:\n", b)

File <__array_function__ internals>:180, in sum(*args, **kwargs)

File ~/.conda/envs/ivy/lib/python3.9/site-packages/numpy/core/fromnumeric.py:2298, in sum(a, axis, dtype, out, keepdims, initial, where)
   2295         return out
   2296     return res
-> 2298 return _wrapreduction(a, np.add, 'sum', axis, dtype, out, keepdims=keepdims,
   2299                       initial=initial, where=where)

File ~/.conda/envs/ivy/lib/python3.9/site-packages/numpy/core/fromnumeric.py:86, in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     83         else:
     84             return reduction(axis=axis, out=out, **passkwargs)
---> 86 return ufunc.reduce(obj, axis, dtype, out, **passkwargs)

ValueError: reduction operation 'add' does not have an identity, so to use a where mask one has to specify 'initial'
```

Reply:
The error is caused by a mistake in the data generation, your data generation allow initial to be None while where is not None. So by adding a check to set where to none when initial is generated to be none will solve the problem. Thank you.

Reply:
```
import numpy as np


a = np.arange(8).reshape((2, 2, 2))
b = np.sum(a, keepdims=False, axis=0)
b = np.sum(a, keepdims=False, axis=0, initial=np._NoValue, where=np._NoValue)
b = np.sum(a, keepdims=False, axis=0, initial=None, where=None)

```

I think you maybe need to have a look at this code and test it.

Reply:
Well then, May be you can update the back end implementation the function in ivy code base to allow it to convert None to np._NoValue. Because we are not supposed to include any dependency of numpy in the front end implementations as this should be able to run on any backend framwork settings. Thank you.

Reply:
Done, convert **None**  to **np._NoValue**. However, in the test, helpers.py, it will call the ground function to make the value test, so in the test case, we need to use **np._NoValue**, or, it will cause error for the **None** when there is not **initial**, but exist **where**